generic-map.js

```diff
@@ -159,6 +160,14 @@ GenericMap.prototype.entries = function () {
     });
 };
 
+GenericMap.prototype.iterate = function() {
+    // v.1 iterator
+    var self = this;
+    return new Iterator(function() {
+        return self.iterator.next().value;
```

Comment:
Where does the iterator property come from? Can more than one iterator walk a given map simultaneously?